python/ray/_private/structured_logging/filters.py

```diff
@@ -0,0 +1,28 @@
+import logging
+import ray
+
+from enum import Enum
+
+
+class CoreContextEnum(str, Enum):
+    JOB_ID = "job_id"
+    WORKER_ID = "worker_id"
+    NODE_ID = "node_id"
+    ACTOR_ID = "actor_id"
+    TASK_ID = "task_id"
+
+
+class CoreContextFilter(logging.Filter):
+    def filter(self, record):
+        runtime_context = ray.get_runtime_context()
+        setattr(record, CoreContextEnum.JOB_ID, runtime_context.get_job_id())
+        setattr(record, CoreContextEnum.WORKER_ID, runtime_context.get_worker_id())
+        setattr(record, CoreContextEnum.NODE_ID, runtime_context.get_node_id())
+        if runtime_context.worker.mode == ray.WORKER_MODE:
```

Comment:
Was there a reason we don't want to set actor id and task id outside of worker mode? Do they not exist in those other cases? Would those throw error or just return None? Maybe we don't need this if statement if they are just going to be None?

Reply:
A warning message will be printed https://sourcegraph.com/github.com/ray-project/ray@d0ed63fad6c80d7d2f7425aafd76c44cd9e5dde2/-/blob/python/ray/runtime_context.py?L254 if we call `get_actor_id` / `get_task_id` in a non-worker process.